packages/grafbase-graphiql/src/playground.tsx

```diff
@@ -17,58 +17,54 @@ import { isLiveQuery, SSEProvider, useSSEContext } from './utils/sse'
 import { getStorage } from './utils/storage'
 import { validateQuery } from './utils/validate-query'
 
-type BaseProps = {
-  logo?: React.ReactNode
-}
-
-type InterfaceProps = GraphiQLInterfaceProps & BaseProps
-
-const Interface = (props: InterfaceProps) => {
-  const { logo, ...rest } = props
-
-  return (
-    <GraphiQLInterface {...rest}>
-      <GraphiQL.Logo>{logo ?? <GrafbaseLogo />}</GraphiQL.Logo>
-      <GraphiQL.Toolbar>
-        <Toolbar />
-      </GraphiQL.Toolbar>
-    </GraphiQLInterface>
-  )
-}
-
-type ProviderProps = Omit<GraphiQLProviderProps, 'children'> & BaseProps
+type PlaygroundProps = Omit<
+  GraphiQLProviderProps,
+  'children' | 'fetcher' | 'storage'
+> &
+  GraphiQLInterfaceProps & {
+    endpoint: string
+    storageKey?: string
+    logo?: ReactNode
+  }
 
-const Provider = (props: ProviderProps) => {
-  const { logo, ...rest } = props
-  return (
-    <GraphiQLProvider
-      {...rest}
-      onTabChange={(tabsState) => {
-        const tabNames = tabsState.tabs.map((tab) => tab.title)
-        setTimeout(() => renameTabs(tabNames), 0)
-        props.onTabChange?.(tabsState)
-      }}
-    >
-      <Interface isHeadersEditorEnabled={false} logo={logo} />
-    </GraphiQLProvider>
-  )
-}
-
-type PlaygroundProps = Omit<ProviderProps, 'fetcher' | 'storage'> & {
-  endpoint: string
-  storageKey?: string
-}
-
-const Playground = (props: PlaygroundProps) => {
-  const { storageKey = 'grafbase', endpoint, ...rest } = props
+const Playground = ({
+  storageKey = 'grafbase',
+  endpoint,
+  logo,
+  dangerouslyAssumeSchemaIsValid,
+  defaultQuery,
+  defaultTabs,
+  externalFragments,
+  getDefaultFieldNames,
+  headers,
+  initialTabs,
+  inputValueDeprecation,
+  introspectionQueryName,
+  maxHistoryLength,
+  onEditOperationName,
+  onSchemaChange,
+  onTabChange,
+  onTogglePluginVisibility,
+  operationName,
+  plugins,
+  query,
+  response,
+  schema,
+  schemaDescription,
+  shouldPersistHeaders,
+  validationRules,
+  variables,
+  visiblePlugin,
+  defaultHeaders,
+  ...props
+}: PlaygroundProps) => {
   const { sseFetcher } = useSSEContext()
 
-  const headers: Record<string, string> | undefined = props.headers
-    ? JSON.parse(props.headers)
-    : undefined
-
   const getFetcher = useCallback<Fetcher>(
     (graphQLParams, fetcherOpts) => {
+      const _headers: Record<string, string> | undefined = headers
```

Comment:
Why not directly const headers, so you don't need to repeat the code at the bottom`{ url: endpoint as string, headers`}

Reply:
`headers` is already defined above 
And parsed headers (`_headers`) are only used in getFetcher function

Reply:
Check the latest changes - does it look more readable now?

Reply:
Nice!